packages/build-tools/src/common/projectSources.ts

```diff
@@ -0,0 +1,110 @@
+import path from 'path';
+
+import spawn from '@expo/turtle-spawn';
+import fs from 'fs-extra';
+import { ArchiveSourceType, Job } from '@expo/eas-build-job';
+import { bunyan } from '@expo/logger';
+import downloadFile from '@expo/downloader';
+
+import { BuildContext } from '../context';
+
+export async function prepareProjectSourcesAsync<TJob extends Job>(
+  ctx: BuildContext<TJob>
+): Promise<void> {
+  if ([ArchiveSourceType.S3, ArchiveSourceType.GCS].includes(ctx.job.projectArchive.type)) {
+    throw new Error('GCS and S3 project sources should be resolved earlier to url');
+  } else if (ctx.job.projectArchive.type === ArchiveSourceType.PATH) {
+    await prepareProjectSourcesLocallyAsync(ctx, ctx.job.projectArchive.path); // used in eas build --local
+  } else if (ctx.job.projectArchive.type === ArchiveSourceType.URL) {
+    await downloadAndUnpackProjectFromTarGzAsync(ctx, ctx.job.projectArchive.url);
+  } else if (ctx.job.projectArchive.type === ArchiveSourceType.GIT) {
+    await shallowCloneRepositoryAsync(
+      ctx,
+      ctx.job.projectArchive.repositoryUrl,
+      ctx.job.projectArchive.gitRef
+    );
+  }
+}
+
+async function shallowCloneRepositoryAsync<TJob extends Job>(
+  ctx: BuildContext<TJob>,
+  projectRepoUrl: string,
+  gitRef: string
+): Promise<void> {
+  try {
+    await spawn('git', ['init'], { cwd: ctx.buildDirectory });
+    await spawn('git', ['remote', 'add', 'origin', projectRepoUrl], { cwd: ctx.buildDirectory });
+    await spawn('git', ['fetch', 'origin', '--depth', '1', gitRef], { cwd: ctx.buildDirectory });
+    await spawn('git', ['checkout', gitRef], { cwd: ctx.buildDirectory });
+  } catch (err: any) {
+    const sanitizedUrl = getSanitizedGitUrl(projectRepoUrl);
+    if (sanitizedUrl) {
+      ctx.logger.error('Failed to clone git repository.');
+    } else {
+      ctx.logger.error(`Failed to clone git repository: ${sanitizedUrl}.`);
+    }
```

Comment:
Shouldn't this be:
```suggestion
    if (!sanitizedUrl) {
      ctx.logger.error('Failed to clone git repository.');
    } else {
      ctx.logger.error(`Failed to clone git repository: ${sanitizedUrl}.`);
    }
```
?